### DISCUSSION TITLE
Create resource with multiple action button

### DISCUSSION THREAD

**Discussion by dityaaa** [filamentphp/filament#4123] [Help] [answered]
I have a form that is used for resource creation. Here I gave 2 main buttons that serve as a submit button. Those 2 buttons should be mutating form state (is_published) based on user action. The thing is, mutateFormDataUsing does not called when action is called. Are there any steps that I do wrong?

![image](https://user-images.githubusercontent.com/25661768/191474558-b4da9280-b69f-4be5-b0e2-07574e0925af.png)

```php
    protected function getSavePublishFormAction(): Action
    {
        return Action::make('publish')
            ->label('Save & Publish')
            ->mutateFormDataUsing(function ($data) {
                $data['is_published'] = true;
                return $data;
            })
            ->submit('create');
    }

    protected function getSaveDraftFormAction(): Action
    {
        return Action::make('draft')
            ->label('Save & Draft')
            ->mutateFormDataUsing(function ($data) {
                $data['is_published'] = false;
                return $data;
            })
            ->color('secondary')
            ->submit('create');
    }
```

**Comment by danharrin** [comment, marked as answer]
This is what we do on the Filament website right now for plugin submissions - https://github.com/filamentphp/filamentphp.com/blob/main/app/Filament/Resources/PluginResource/Pages/CreatePlugin.php#L20-L62

**Reply by Cepairda** [reply]
I have this problen in 2025 year
